<launch>

  <arg name="world_path" default="$(find uavros_gazebo)/worlds/precision_landing.world" />
  <arg name="vehicle" default="iris" />

  <include file="$(find uavros_gazebo)/launch/mavros_gazebo_px4_sitl.launch">
    <arg name="world" value="$(arg world_path)" />
    <arg name="vehicle" value="$(arg vehicle)" />   
  </include>

    <node pkg="uavros_uavugv_sitl" type="uav_controller_node" name="uav_controller" output="screen">
        <param name="alt_sp" value = "2.5"/>
    	  <param name="Kp_x" value = "2"/>
        <param name="Kp_y" value = "2"/>
        <param name="Kv_x" value = "0.7"/>
        <param name="Kv_y" value = "0.7"/>
    </node>

    <node pkg="uavros_uavugv_sitl" type="leader_estimator_node" name="leader_estimator" output="screen">
        <param name="shape_omega" value = "0.5"/>
    	  <param name="shape_radius" value = "2.0"/>
        <param name="alt_sp" value = "2.5"/>
    </node>

  
  <arg name="visualization" default="true"/>
  <group if="$(arg visualization)">
      <node type="rviz" name="rviz" pkg="rviz" args="-d $(find uavros_uavugv_sitl)/config/rviz_single.rviz" />
  </group>

</launch>